### DISCUSSION TITLE
Anyway to Ensure GitHub Actions Jobs Are Running in Container on Self-Hosted Runners?

### DISCUSSION THREAD

**Discussion by ukewea** [community/community#123572] [Actions] [answered]
### Select Topic Area

Question

### Body

**Update:**

If anyone has similar requirement like me, just add  `ACTIONS_RUNNER_REQUIRE_JOB_CONTAINER=true`  to the `.env` file in your Actions Runner's installation directory (as @ChristopherHX pointed out).

For example, if my Actions Runner is in `/data/actions-runner`, just modify `/data/actions-runner/.env`, add  a new line with text `ACTIONS_RUNNER_REQUIRE_JOB_CONTAINER=true`, then restart the runner process. 

No pre-run scripts required.

---

Hi everyone,

I'm running a self-hosted GitHub Actions runner and need every job to run inside a container. I discovered the feature that allows [running a script before or after a job](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job) and planned to use it to stop any job not running in a container.

I implemented a script to run on job started and defined an environment hook as follows:

```bash
ACTIONS_RUNNER_HOOK_JOB_STARTED=/data/actions-runner-scripts/on_job_started.sh
```

However, it turns out **the job runs when the job starts but before the container is created**, as shown in the screenshot below, so there's no way to do the detection with this hook (as the container is not even started yet).

![Screenshot 2024-05-13 140130](https://github.com/community/community/assets/60734042/d90b6eeb-086c-421e-9127-8ba47c838f07)

Is there any event where I can inject my script to ensure all jobs running on the runner all execute inside a container? or I should use other mechanism(s) achieve this?

Thanks.

---

For anyone interested, here's the script I used:

```bash
#!/bin/bash

# Check if the script is running inside a Docker container by checking for /.dockerenv
if [ ! -f /.dockerenv ]; then
  echo "::error ::This script must be run in a Docker container"
  exit 1
fi

echo "Running inside a Docker container. Proceeding with the script..."
```

**Comment by Saad-Ayady** [comment]
1

**Comment by ChristopherHX** [comment, marked as answer]
That's specfic usecase is covered by `ACTIONS_RUNNER_REQUIRE_JOB_CONTAINER` see [here](https://github.com/actions/runner-container-hooks/blob/main/packages/k8s/README.md#pre-requisites)

> The ACTIONS_RUNNER_REQUIRE_JOB_CONTAINER env should be set to true to prevent the runner from running any jobs outside of a container

**Reply by ukewea** [reply]
Thanks @ChristopherHX, adding this environment variable to `.env` perfectly fits our need. I will update the original question for future readers' convenience.

![image](https://github.com/community/community/assets/60734042/ea104afe-40de-4b50-8df1-a447c4faaba4)
